packer/rpc/communicator.go

```diff
@@ -97,7 +112,9 @@ func (c *communicator) Start(cmd *packer.RemoteCmd) (err error) {
 
 		var finished CommandFinished
 		decoder := gob.NewDecoder(conn)
-		if err := decoder.Decode(&finished); err != nil {
+		err = decoder.Decode(&finished)
+		wg.Wait()
```

Comment:
I have at least one person on the original issue (the OP) who's lost output when erroring out of a shell script, I imagine because of the `cmd.SetExited(123)`; I think waiting here is the safest option.